### DISCUSSION TITLE
Building for Deployment

### DISCUSSION THREAD

**Discussion by habitchrisl** [holger1411/understrap#2197] [Q&A] [answered]
Hey there, I was using an really old version of Understrap that was modified by someone else at my place of work, of whom has left ages, but have decided to switch over to a more recent version - since Gulp is no longer recommended, I've been using the default ``npm run dist-build`` when deploying to production via a DeployHQ pipeline.

I have noticed that the ``src`` folder still exists and the ``dist`` folder is also created, but I'm not sure if I understanding this right. Should the ``src`` folder and everything in the child theme's root directory still be there? Should it not just be the contents of the ``dist`` folder?  Thanks for the clarification!

**Comment by bacoords** [comment, marked as answer]
So in the current version of Understrap, the way it works is:

`dist` - builds just the assets, something you'd want to use in a CI/CD environment, especially if you're not version controlling the built assets. The built assets just show up in the `/css` and `/js` directories on the root.

`dist-build` - builds a separate `dist` directory with a copy of the entire theme without source files and other build process-related files. mainly for use in distributing as a Zip file, like on the WordPress.org theme directory.

**Reply by habitchrisl** [reply]
Ah I see, perhaps I want to be using ```npm run dist``` then if I'm deploying the website myself? The issue I have with that is, wouldn't it also include all the dev folders such as ```node_modules``` and such?

**Reply by bacoords** [reply]
It will include some stuff you don't need, like package.json. Node modules should be gitignored though and not stored anywhere. A lot of it depends on your particular deployment setup. For example, you some tools can have a separate deployignore file where you can tell it not to push certain files to your server

**Reply by habitchrisl** [reply]
Interesting, I wonder why node_modules folder is being shoved in there for me then. It must be a config somewhere as you say. Thanks!
